access/access.go

```diff
@@ -0,0 +1,144 @@
+package access
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/qlik-oss/enigma-go"
+)
+
+const (
+	trafficDumpFile = "traffic-dump.json"
+)
+
+var (
+	secret = []byte("secret")
+	ctx    = context.Background()
+)
+
+// connect connects to an engine at the proviced URL and claims.
+func connect(engineURL string, jwtClaims jwt.MapClaims) (*enigma.Global, error) {
+	ctx := context.Background()
+	headers := make(http.Header)
+
+	if jwtClaims != nil {
+		signedJwt, _ := jwt.NewWithClaims(jwt.SigningMethodHS256, jwtClaims).SignedString(secret)
+		headers.Set("Authorization", fmt.Sprintf("Bearer %s", signedJwt))
+	}
+
+	return enigma.Dialer{TrafficDumpFile: trafficDumpFile}.Dial(ctx, engineURL, headers)
+}
+
+// createApp creates an app using the provided Global object and app name.
+func createApp(global *enigma.Global, appName string) (*enigma.Doc, error) {
+	return global.CreateDocEx(ctx, appName, "", "", "", "Main")
+}
+
+// openApp opens an app using the providad Global object and app name.
+func openApp(global *enigma.Global, appID string) (*enigma.Doc, error) {
+	return global.OpenDoc(ctx, appID, "", "", "", false)
+}
+
+// createSheet creates a sheet object using the provided Doc objecgt and sheet name.
+func createSheet(doc *enigma.Doc, name string) (*enigma.GenericObject, error) {
+	msg := json.RawMessage(`{
+		"title": "/title",
+		"description": "/description",
+		"meta": "/meta",
+		"order": "/order",
+		"type": "/qInfo/qType",
+		"id": "/qInfo/qId",
+		"lb": "/qListObjectDef",
+		"hc": "/qHyperCubeDef"
+	}`)
+
+	props := enigma.GenericObjectProperties{
+		Info: &enigma.NxInfo{Id: name, Type: "sheet"},
```

Comment:
I'd prefer if we keep the example generic, avoiding examples of how to do access rules on Sense-specific objects (type 'sheet', which has an 'order' property)